conn.go

```diff
@@ -23,19 +24,57 @@ func beginHandlerOpen(ch chan error, removeHandlers func()) func(ev *js.Object)
 	}
 }
 
+// closeErrorMap maps Close frame status codes to their string explanation
+// See http://tools.ietf.org/html/rfc6455#section-7.4.1
+var closeErrorMap = map[int]string{
+	1000: "Normal closure, meaning that the purpose for which the connection was established has been fulfilled.",
+	1001: "An endpoint is \"going away\", such as a server going down or a browser having navigated away from a page.",
+	1002: "An endpoint is terminating the connection due to a protocol error",
+	1003: "An endpoint is terminating the connection because it has received a type of data it cannot accept (e.g., an endpoint that understands only text data MAY send this if it receives a binary message).",
+	1004: "Reserved. The specific meaning might be defined in the future.",
+	1005: "No status code was actually present.",
+	1006: "The connection was closed abnormally, e.g., without sending or receiving a Close control frame",
+	1007: "An endpoint is terminating the connection because it has received data within a message that was not consistent with the type of the message (e.g., non-UTF-8 [http://tools.ietf.org/html/rfc3629] data within a text message).",
+	1008: "An endpoint is terminating the connection because it has received a message that \"violates its policy\". This reason is given either if there is no other sutible reason, or if there is a need to hide specific details about the policy.",
+	1009: "An endpoint is terminating the connection because it has received a message that is too big for it to process.",
+	// Note that this status code is not used by the server, because it can fail the WebSocket handshake instead
+	1010: "An endpoint (client) is terminating the connection because it has expected the server to negotiate one or more extension, but the server didn't return them in the response message of the WebSocket handshake.",
+	1011: "A server is terminating the connection because it encountered an unexpected condition that prevented it from fulfilling the request.",
+	1015: "The connection was closed due to a failure to perform a TLS handshake (e.g., the server certificate can't be verified).",
+}
+
 // closeError allows a CloseEvent to be used as an error.
 type closeError struct {
 	*dom.CloseEvent
 }
 
 func (e *closeError) Error() string {
-	var cleanStmt string
+	var (
+		cleanStmt string
+		reason    string // cant override the event's reason
+		ok        bool
+	)
 	if e.WasClean {
 		cleanStmt = "clean"
 	} else {
 		cleanStmt = "unclean"
+
+		if e.Reason != "" {
+			reason = e.Reason
+		} else {
+			reason, ok = closeErrorMap[e.Code]
+			if !ok {
+				reason = "Unknown reason"
+			}
+
+			if e.Code == 1010 {
+				reason += "\nSpecifically, the extensions that are needed are: " + e.Reason
```

Comment:
Oh right, that does nothing. Thanks @shurcooL.

Sadly I havn't seen a websocket implementation (on browser api level) that fills in those fields to acutally see how concatanating these would makes sense. In this case, if `e.Reason` has more than a list of exensions in it, this would get ugly. I'll remove the `if e.Code == 1010 {}` block until then.